common-lib/src/main/scala/com/gu/mediaservice/lib/ImageIngestOperations.scala

```diff
@@ -20,19 +20,43 @@ class ImageIngestOperations(imageBucket: String, thumbnailBucket: String, config
 
   import ImageIngestOperations.{fileKeyFromId, optimisedPngKeyFromId}
 
-  def storeOriginal(id: String, file: File, mimeType: Option[MimeType], meta: Map[String, String] = Map.empty)
-                   (implicit logMarker: LogMarker): Future[S3Object] =
-    storeImage(imageBucket, fileKeyFromId(id), file, mimeType, meta)
-
-  def storeThumbnail(id: String, file: File, mimeType: Option[MimeType])
-                    (implicit logMarker: LogMarker): Future[S3Object] =
-    storeImage(thumbnailBucket, fileKeyFromId(id), file, mimeType)
-
-  def storeOptimisedPng(id: String, file: File)
+  def store(storableImage: StorableImage)
+           (implicit logMarker: LogMarker): Future[S3Object] = storableImage match {
+    case s:StorableOriginalImage => storeOriginalImage(s)
+    case s:StorableThumbImage => storeThumbnailImage(s)
+    case s:StorableOptimisedImage => storeOptimisedImage(s)
+  }
+
+  private def storeOriginalImage(storableImage: StorableImage)
+                        (implicit logMarker: LogMarker): Future[S3Object] =
+    storeImage(imageBucket, fileKeyFromId(storableImage.id), storableImage.file, Some(storableImage.mimeType), storableImage.meta)
+
+  private def storeThumbnailImage(storableImage: StorableImage)
+                         (implicit logMarker: LogMarker): Future[S3Object] =
+    storeImage(thumbnailBucket, fileKeyFromId(storableImage.id), storableImage.file, Some(storableImage.mimeType))
+
+  private def storeOptimisedImage(storableImage: StorableImage)
                        (implicit logMarker: LogMarker): Future[S3Object] =
-    storeImage(imageBucket, optimisedPngKeyFromId(id), file, Some(Png))
-  
+    storeImage(imageBucket, optimisedPngKeyFromId(storableImage.id), storableImage.file, Some(storableImage.mimeType))
+
   def deleteOriginal(id: String): Future[Unit] = if(isVersionedS3) deleteVersionedImage(imageBucket, fileKeyFromId(id)) else deleteImage(imageBucket, fileKeyFromId(id))
   def deleteThumbnail(id: String): Future[Unit] = deleteImage(thumbnailBucket, fileKeyFromId(id))
   def deletePng(id: String): Future[Unit] = deleteImage(imageBucket, optimisedPngKeyFromId(id))
 }
+
+
+
+trait StorableImage {
+  val id: String
+  val file: File
+  val mimeType: MimeType
+  val meta: Map[String, String]
+}
+case class StorableThumbImage(id: String, file: File, mimeType: MimeType, meta: Map[String, String] = Map.empty) extends StorableImage
```

Comment:
We've refactored here to codify the three possible sorts of images in types. In this way, it's harder to make an error by mistaking one `StorableImage` for another.